osu.Game/Screens/Ranking/FavouriteButton.cs

```diff
@@ -0,0 +1,164 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+#nullable disable
+
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Logging;
+using osu.Game.Beatmaps;
+using osu.Game.Beatmaps.Drawables.Cards;
+using osu.Game.Graphics;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Online.API;
+using osu.Game.Online.API.Requests;
+using osu.Game.Online.API.Requests.Responses;
+using osu.Game.Resources.Localisation.Web;
+using osuTK;
+
+namespace osu.Game.Screens.Ranking
+{
+    public partial class FavouriteButton : OsuAnimatedButton
```

Comment:
There's now three versions of this code. This doesn't look to be copy-pasted from either other implementation. Did you miss these somehow? If so, please search for existing implementations before reimplementing.

If there's a reason for the different implementation, please let us know.

Reply:
I originally copy pasted it from the `osu.Game/Overlays/BeatmapSet/Buttons/FavouriteButton.cs` file, but that takes in an `APIBeatmapSet` object which doesn't exist on the rankings screen. 

The `BeatmapSet` version also doesn't handle signing into another account either (favorited on account A, but not on account B [or signs out -> signs back in]) since the page gets "reloaded" as the user signs out while the rankings screen does not get reloaded.

My original implementation in https://github.com/ppy/osu/commit/0bc14ba646e8fc7adfa9d8e3b53d0bf1341239e0 was more similar to the `BeatmapSet` version, but it had content shifting as the button loaded, and would load after a long time on slower internet; hence why I went forward with the reimplementation.

Reply:
> The `BeatmapSet` version also doesn't handle signing into another account either (favorited on account A, but not on account B [or signs out -> signs back in]) since the page gets "reloaded" as the user signs out while the rankings screen does not get reloaded.


https://github.com/user-attachments/assets/35d68c78-fc22-4103-8df8-c552a4b9c22e

It also works if a guest user checks the rankings screen